pkg/asset/agent/agentconfig/agent_config.go

```diff
@@ -11,6 +11,7 @@ import (
 	"sigs.k8s.io/yaml"
 
 	"github.com/openshift/installer/pkg/asset"
+	"github.com/openshift/installer/pkg/asset/agent/joiner"
```

Comment:
It's a bit weird for the agent-config asset to be depending on the node-joiner, and will only get weirder when IPI needs to use it as well. I wonder if this code should live in `pkg/types/baremetal`, or failing that perhaps somewhere under `pkg/util`.